contracts/service-provider-directory/src/state.rs

```diff
@@ -0,0 +1,176 @@
+use cosmwasm_std::{Addr, Coin, StdResult, Storage, Uint128};
+use cw_storage_plus::{Item, Map};
+use serde::{Deserialize, Serialize};
+
+// Storage keys
+pub const CONFIG_KEY: &str = "config";
+pub const SERVICE_ID_COUNTER_KEY: &str = "sidc";
+pub const SERVICES_KEY: &str = "services";
+
+// Storage
+pub const CONFIG: Item<Config> = Item::new(CONFIG_KEY);
+pub const SERVICES: Map<ServiceId, Service> = Map::new(SERVICES_KEY);
+pub const SERVICE_ID_COUNTER: Item<ServiceId> = Item::new(SERVICE_ID_COUNTER_KEY);
+
+/// The directory of services are indexed by [`ServiceId`].
+pub type ServiceId = u64;
+
+/// The type of services provider supported
+#[derive(Serialize, Deserialize, Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ServiceType {
+    NetworkRequester,
+}
+
+impl std::fmt::Display for ServiceType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let service_type = match self {
+            ServiceType::NetworkRequester => "network_requester",
+        };
+        write!(f, "{service_type}")
+    }
+}
+
+/// The types of addresses supported.
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub enum NymAddress {
```

Comment:
It kinda feels like those types would like to live outside the contract, maybe in some `service-provider-directory-common` crate?